src/EnergyPlus/api/runtime.cc

```diff
@@ -77,261 +77,265 @@ int energyplus(EnergyPlusState state, int argc, const char *argv[])
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     thisState->dataGlobal->stopSimulation = true;
 }
 
 void setConsoleOutputState(EnergyPlusState state, int outputStatus)
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
-    if (outputStatus == 0) {
-        thisState->dataGlobal->printConsoleOutput = false;
-    } else {
-        thisState->dataGlobal->printConsoleOutput = true;
-    }
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    thisState->dataGlobal->printConsoleOutput = (outputStatus != 0);
 }
 
 void setEnergyPlusRootDirectory(EnergyPlusState state, const char *path)
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     thisState->dataGlobal->installRootOverride = true;
     thisState->dataStrGlobals->exeDirectoryPath = fs::path(path);
 }
 
 void issueWarning(EnergyPlusState state, const char *message)
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     EnergyPlus::ShowWarningError(*thisState, message);
 }
 void issueSevere(EnergyPlusState state, const char *message)
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     EnergyPlus::ShowSevereError(*thisState, message);
 }
 void issueText(EnergyPlusState state, const char *message)
 {
-    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     EnergyPlus::ShowContinueError(*thisState, message);
 }
 
+void registerProgressCallback([[maybe_unused]] EnergyPlusState state, std::function<void(int const)> f)
+{
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    thisState->dataGlobal->progressCallback = std::move(f);
+}
+
 void registerProgressCallback(EnergyPlusState state, void (*f)(int const))
 {
-    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    auto *thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
     thisState->dataGlobal->progressCallback = f;
 }
 
 void registerStdOutCallback([[maybe_unused]] EnergyPlusState state, std::function<void(const std::string &)> f)
```

Comment:
@Myoldmopar Right back at you. Why is this marked `[[maybe_unused]]`? It definitely appears used.